src/HealthChecks/HealthChecks/src/HealthCheckPublisherHostedService.cs

```diff
@@ -110,6 +150,28 @@ internal void CancelToken()
         _runTokenSource!.Cancel();
     }
 
+    private async Task CheckHealthAsync(string healthCheckName)
+    {
+        // Concatenate with exposed predicate
+        var runHealthCheckPredicate = (HealthCheckRegistration hcr) =>
+        {
+            var runHealthCheck = hcr.Name == healthCheckName;
```

Comment:
This seems really inefficient. It used to run all checks in one pass on one timer event. Now it has N timer events and in each it filters the list of checks to one and runs that.

What if we considered more significant changes to HealthCheckService to have it schedule the checks instead? Then when it was told to CheckHealth, it could return the latest results, and run any checks that didn't have a period specified?